libraries/3rdparty/mbedtls_utils/mbedtls_error.c

```diff
@@ -220,1364 +220,1117 @@ const char * mbedtls_strerror_highlevel( int errnum )
+        #endif /* MBEDTLS_PLATFORM_C */
+
+        #if defined( MBEDTLS_POLY1305_C )
+            case -( MBEDTLS_ERR_POLY1305_BAD_INPUT_DATA ):
+                rc = "POLY1305 - Invalid input parameter(s)";
+                break;
+
+            case -( MBEDTLS_ERR_POLY1305_FEATURE_UNAVAILABLE ):
+                rc = "POLY1305 - Feature not available. For example, s part of the API is not implemented";
+                break;
+
+            case -( MBEDTLS_ERR_POLY1305_HW_ACCEL_FAILED ):
+                rc = "POLY1305 - Poly1305 hardware accelerator failed";
+                break;
+        #endif /* MBEDTLS_POLY1305_C */
+
+        #if defined( MBEDTLS_RIPEMD160_C )
+            case -( MBEDTLS_ERR_RIPEMD160_HW_ACCEL_FAILED ):
+                rc = "RIPEMD160 - RIPEMD160 hardware accelerator failed";
+                break;
+        #endif /* MBEDTLS_RIPEMD160_C */
+
+        #if defined( MBEDTLS_SHA1_C )
+            case -( MBEDTLS_ERR_SHA1_HW_ACCEL_FAILED ):
+                rc = "SHA1 - SHA-1 hardware accelerator failed";
+                break;
+
+            case -( MBEDTLS_ERR_SHA1_BAD_INPUT_DATA ):
+                rc = "SHA1 - SHA-1 input data was malformed";
+                break;
+        #endif /* MBEDTLS_SHA1_C */
+
+        #if defined( MBEDTLS_SHA256_C )
+            case -( MBEDTLS_ERR_SHA256_HW_ACCEL_FAILED ):
+                rc = "SHA256 - SHA-256 hardware accelerator failed";
+                break;
+
+            case -( MBEDTLS_ERR_SHA256_BAD_INPUT_DATA ):
+                rc = "SHA256 - SHA-256 input data was malformed";
+                break;
+        #endif /* MBEDTLS_SHA256_C */
+
+        #if defined( MBEDTLS_SHA512_C )
+            case -( MBEDTLS_ERR_SHA512_HW_ACCEL_FAILED ):
+                rc = "SHA512 - SHA-512 hardware accelerator failed";
+                break;
+
+            case -( MBEDTLS_ERR_SHA512_BAD_INPUT_DATA ):
+                rc = "SHA512 - SHA-512 input data was malformed";
+                break;
+        #endif /* MBEDTLS_SHA512_C */
+
+        #if defined( MBEDTLS_THREADING_C )
+            case -( MBEDTLS_ERR_THREADING_FEATURE_UNAVAILABLE ):
+                rc = "THREADING - The selected feature is not available";
+                break;
+
+            case -( MBEDTLS_ERR_THREADING_BAD_INPUT_DATA ):
+                rc = "THREADING - Bad input parameters to function";
+                break;
+
+            case -( MBEDTLS_ERR_THREADING_MUTEX_ERROR ):
+                rc = "THREADING - Locking / unlocking / free failed with error code";
+                break;
+        #endif /* MBEDTLS_THREADING_C */
+
+        #if defined( MBEDTLS_XTEA_C )
+            case -( MBEDTLS_ERR_XTEA_INVALID_INPUT_LENGTH ):
+                rc = "XTEA - The data input has an invalid length";
+                break;
+
+            case -( MBEDTLS_ERR_XTEA_HW_ACCEL_FAILED ):
+                rc = "XTEA - XTEA hardware accelerator failed";
+                break;
+        #endif /* MBEDTLS_XTEA_C */
+    }
```

Comment:
Should there be a `default:` for this `switch`?

Reply:
Good question. The default value will continue to be the `NULL` initialized value of the return variable in this case.